### DISCUSSION TITLE
Can TEN Agent digest my knowledge base documentation

### DISCUSSION THREAD

**Discussion by swloo618** [TEN-framework/ten-framework#345] [General]
May I know TEN agent able to digest my knowledge base and answer the question accordingly.  How I can let TEN agent digest it?

**Comment by hemantrao12** [comment]
We would have to modify the code in order to have the access to GPT assistants or gemini knowledge base

**Reply by adlermore** [reply]
To integrate GPT assistants or Gemini knowledge base, you need to obtain API access (e.g., OpenAI for GPT or Google's Gemini). Once you have API credentials, you can connect your app by sending requests to the API and receiving responses. Handle these responses according to your app's needs, such as displaying them in a chat or using them for specific actions.

**Comment by plutoless** [comment]
Modify the code or creating an extension is definitely a way. But with our latest release of a dify extension, there's one more way you can achieve this by using tools like Dify.
You can import your knowledge base docs into your chat bot created in dify, and use Ten to make your chatbot listen and speak in a second. Check the tutorial [here](https://doc.theten.ai/ten-agent/quickstart-1/run_dify).

**Comment by hemantrao12** [comment]
I am getting only these options but not dify. 
<img width="316" alt="Screenshot 2024-12-26 at 12 35 23 PM" src="https://github.com/user-attachments/assets/44032d0b-f727-4db7-8d84-082787c1194a" />

**Reply by plutoless** [reply]
when selecting graph, choose voice assistant stt integrated / voice assistant
realtime currently does not support RAG well.
